### PR TITLE
Update app platform docs to use Catalog CRD

### PR DESCRIPTION
**Pull request by rossf7**
Follow up to https://github.com/giantswarm/docs/pull/1001 to use the new CRD in the app platform section of our docs.

### Please remember to

- [x] Give the PR a meaningful title -- it will be shown in the release notes
- [ ] Add (or remove) [user questions](https://github.com/giantswarm/docs/blob/master/CONTRIBUTING.md#front-matter) associated with any updated docs

### CODE REVIEW

**Review by marians — APPROVED** [review]
LGTM by a quick check

File: src/content/app-platform/defaulting-validation/index.md [hunk 1]
```diff
@@ -111,10 +111,10 @@ Currently we validate:
 
 - The `app-operator.giantswarm.io/version` label is present.
 - All referenced configmaps and secrets exist.
-- The catalog has a matching AppCatalog CR.
+- The catalog has a matching Catalog CR.
 - There is no App CR with a conflicting annotation or label value for the [target namespace]({{< relref "/app-platform/namespace-configuration" >}}).
 
-If app-operator finds a matching [AppCatalogEntry]({{< relref "/ui-api/management-api/crd/appcatalogentries.application.giantswarm.io.md" >}}) CR, it will use this to run more validation checks.
+If app-operator finds a matching [CatalogEntry]({{< relref "/ui-api/management-api/crd/appcatalogentries.application.giantswarm.io.md" >}}) CR, it will use this to run more validation checks.
```
Comment (marians) [inline src/content/app-platform/defaulting-validation/index.md:117]: Wait, the short name changes from AppCatalogEntry to CatalogEntry, but the CRD link stays the same?

Comment (rossf7) [reply src/content/app-platform/defaulting-validation/index.md:117]: Damn no that shouldn't change, I'll fix. Thx! Good catch

Comment (rossf7) [reply src/content/app-platform/defaulting-validation/index.md]: I found another problem but looks OK now.

Comment (marians) [reply src/content/app-platform/defaulting-validation/index.md]: Cool

**Pull request merged**